framework/web/Session.php

```diff
@@ -129,6 +131,7 @@ public function open()
         if ($this->getIsActive()) {
             Yii::info('Session started', __METHOD__);
             $this->updateFlashCounters();
+            register_shutdown_function([$this, 'close']);
```

Comment:
It is not very good. In case you are opening and closing session multiple times during the single HTTP request process, such code will produce numerous shutdown functions registration.